Examples/Python/src/Geometry.cpp

```diff
@@ -274,6 +275,37 @@ void addExperimentalGeometry(Context& ctx) {
         .def(py::init<std::vector<std::shared_ptr<Surface>>>());
   }
 
+  {
+    using Range1D = Acts::RangeXD<1u, Acts::ActsScalar>;
+    using KdtSurfaces1D = Acts::Experimental::KdtSurfaces<1u, 100u>;
+    using KdtSurfacesProvider1D =
+        Acts::Experimental::KdtSurfacesProvider<1u, 100u>;
+
+    py::class_<Range1D>(m, "Range1D")
+        .def(py::init([](const std::array<Acts::ActsScalar, 2u>& irange) {
+          Range1D range;
+          range[0].shrink(irange[0], irange[1]);
+          return range;
+        }));
+
+    py::class_<KdtSurfaces1D, std::shared_ptr<KdtSurfaces1D>>(m,
+                                                              "KdtSurfaces1D")
+        .def(py::init<const GeometryContext&,
+                      const std::vector<std::shared_ptr<Acts::Surface>>&,
+                      const std::array<Acts::BinningValue, 1u>&>())
+        .def("surfaces", [](KdtSurfaces1D& self, const Range1D& range) {
+          return self.surfaces(range);
+        });
```

Comment:
```suggestion
        .def("surfaces", &KdtSurfaces1D::surfaces)
```
?

Reply:
Surfaces is a templated method (on the range request), when just doing as you suggest, the template parameter can not be inferred.

Reply:
But the template is already instantiated here, the surfaces is not a method template.

Reply:
I know, it just doesn't compile ... telling me the template can not be inferred

Reply:
Ok, I checked. The problem is that the surfaces method is overloaded, so you need to use `py::overload_cast` to decide which overload you want to expose.

Reply:
Great, thanks!